src/analysis/classic/druid/shared/talents/Revitalize.tsx

```diff
@@ -73,7 +91,11 @@ class Revitalize extends Analyzer {
   }
 
   resourceEventReduce = (totals: ResourcesByPlayer, evt: AnyEvent) => {
-    if (evt.type !== EventType.ResourceChange) {
+    if (
+      evt.type !== EventType.ResourceChange || // Not a resource event
+      !this.eligibleCasts[truncTimestamp(evt)] || // truncated timestamp does not match a rejuv/wild growth heal
+      !this.eligibleCasts[truncTimestamp(evt)][evt.targetID] // no heal for truncated timestamp for the given target
+    ) {
```

Comment:
this can bin two adjacent events into different bins. for example: cast at timestamp 100 and revitalize at 101. those now have different bins and don't match.

your best bet might be storing the eligible casts sorted by timestamp, then doing a binary search for one that is within your tolerance

Reply:
Good idea. If there is a better/existing function let me know, otherwise I have implemented my own.